hledger-lib/Hledger/Data/Journal.hs

```diff
@@ -523,19 +523,21 @@ filterJournalTransactionsByAccount apats j@Journal{jtxns=ts} = j{jtxns=filter tm
 -- use.  Reverse parsed data to normal order, standardise amount
 -- formats, check/ensure that transactions are balanced, and maybe
 -- check balance assertions.
-journalFinalise :: ClockTime -> FilePath -> Text -> Bool -> ParsedJournal -> Either String Journal
-journalFinalise t path txt assrt j@Journal{jfiles=fs} =
-  journalTieTransactions <$> 
-  (journalBalanceTransactions assrt $
-   journalApplyCommodityStyles $
-   j {jfiles        = (path,txt) : reverse fs
-     ,jlastreadtime = t
-     ,jdeclaredaccounts = reverse $ jdeclaredaccounts j
-     ,jtxns         = reverse $ jtxns j         -- NOTE: see addTransaction
-     ,jtxnmodifiers = reverse $ jtxnmodifiers j -- NOTE: see addTransactionModifier
-     ,jperiodictxns = reverse $ jperiodictxns j -- NOTE: see addPeriodicTransaction
-     ,jmarketprices = reverse $ jmarketprices j -- NOTE: see addMarketPrice
-     })
+journalFinalise :: ClockTime -> FilePath -> Text -> Bool -> Bool -> ParsedJournal -> Either String Journal
+journalFinalise t path txt reorder assrt j@Journal{jfiles=fs} =
+  let j' = if reorder
+           then j {jfiles        = (path,txt) : reverse fs
+                  ,jlastreadtime = t
+                  ,jdeclaredaccounts = reverse $ jdeclaredaccounts j
+                  ,jtxns         = reverse $ jtxns j         -- NOTE: see addTransaction
+                  ,jtxnmodifiers = reverse $ jtxnmodifiers j -- NOTE: see addTransactionModifier
+                  ,jperiodictxns = reverse $ jperiodictxns j -- NOTE: see addPeriodicTransaction
+                  ,jmarketprices = reverse $ jmarketprices j -- NOTE: see addMarketPrice
+                  }
+           else j
+  in journalTieTransactions <$>
+     (journalBalanceTransactions assrt $ journalApplyCommodityStyles j')
```

Comment:
Bikeshedding: maybe confining the effect of reorder to a smaller block of code makes it clearer ?
```haskell
journalFinalise t path txt reorder assrt j@Journal { jfiles = fs } =
  journalTieTransactions
  <$> (journalBalanceTransactions assrt $
       journalApplyCommodityStyles j{
          jfiles            = (path, txt) : mayberev fs
          , jlastreadtime     = t
          , jdeclaredaccounts = mayberev $ jdeclaredaccounts j
          , jtxns             = mayberev $ jtxns j         -- NOTE: see addTransaction
          , jtxnmodifiers     = mayberev $ jtxnmodifiers j -- NOTE: see addTransactionModifier
          , jperiodictxns     = mayberev $ jperiodictxns j -- NOTE: see addPeriodicTransaction
          , jmarketprices     = mayberev $ jmarketprices j -- NOTE: see addMarketPrice
          })
  where mayberev = if reorder then reverse else id
```

Reply:
Yeah -- I like this. But in this case, we'd be adding `(path,txt)` to `jfiles` twice in the `--auto` case, right?

Reply:
True. It's also setting jlastreadtime. "reorder" doesn't quite cover what it's about. "finallyfinalise" ? Split this function into two ?

Reply:
I don't see exactly how to clarify it right now.. if you don't see a way, it's ok to leave it how it was.

Reply:
`journalFinalise` is only used in Read.Common's `parseAndFinaliseJournal` (and the primed version). Since it's a single-use function, I wonder if combining all its steps is creating more confusion. I.e., just combine them in `parseAndFinaliseJounal` as we need them, instead of adding a bunch of boolean flags.

Reply:
Perhaps!

Reply:
See 6a4ece6 below for a try at this splitting up (docs aren't added, and it isn't added to the primed function).

Pros: more explicit what's going on; no opaque boolean flags; no running unnecessary steps
Cons: at first blush, a more confusing `parseAndFinaliseJournal` function